initfiles/bin/init_roxie

```diff
@@ -51,6 +54,15 @@ killed() {
     exit 255
 }
 
+# Check for dfsgroup change and throw error/exit
+diff <($INSTALL_DIR/bin/daliadmin server=$DALISERVER dfsgroup $component | sort | uniq) <($INSTALL_DIR/sbin/configgen -env $CONFIG_DIR/$ENV_XML_FILE -c $component -listall | awk 'BEGIN {FS=",";} /RoxieServerProcess/ { print $3 }' | sort | uniq) &>/dev/null
+if [[ $? -ne 0 ]]; then
+  echo "ERROR: dfsgroup $component is not the same as the environment.xml" >> $logfilename.stderr
+  echo "If this was intentional, please run 'updtdalienv ${CONFIG_DIR}/${ENV_XML_FILE} -f'"
+  killed
+fi
```

Comment:
I am not sure Roxie should be prevented from starting up and treated in the same way Thor like this.

Roxie would still start up if the group had changed (but the environment had not) afaik, or have you tested that scenario and know otherwise? 

The reason Thor doesn't startup, is because the group is used and that has not change.
Thor *could* use the environment definition to start up.., but then there'd be potential problems:
a) accessing files on the named group that referred to potentially unreachable IP's, or
b) publishing new files with it's configured named group, which no longer matched the group it was writing to.

It is reasonable to fail to start Thor for this reason and require intervention.

In Roxie's case, it is *generally* reading files created by other engines, so b) is less of a problem (although Roxie can now write files too).
a) is only a problem for locality or if the old IP's are inaccessible.

Either way, I'm not sure we want to fail the startup for Roxie in the same way.
It is probably best to leave Roxie alone for this JIRA/PR and deal with it separately and talk to @richardkchapman about it.